centml/sdk/auth.py

```diff
@@ -0,0 +1,61 @@
+import time
+import sys
+import os
+import json
+import requests
+import jwt
+
+from .config import Config
+
+
+def refresh_centml_token(refresh_token):
+    api_key = Config.firebase_api_key
+
+    cred = requests.post(
+        f"https://securetoken.googleapis.com/v1/token?key={api_key}",
+        headers={"content-type": "application/json; charset=UTF-8"},
+        data=json.dumps({"grantType": "refresh_token", "refreshToken": refresh_token}),
+        timeout=3,
+    ).json()
+
+    with open(Config.centml_cred_file, 'w') as f:
+        json.dump(cred, f)
+
+    return cred
+
+
+def store_centml_cred(token_file):
+    with open(token_file, 'r') as f:
+        os.makedirs(Config.centml_config_dir, exist_ok=True)
+        refresh_token = json.load(f)["refreshToken"]
+
+        refresh_centml_token(refresh_token)
+
+
+def load_centml_cred():
+    cred = None
+
+    if os.path.exists(Config.centml_cred_file):
+        with open(Config.centml_cred_file, 'r') as f:
+            cred = json.load(f)
```

Comment:
for now are we assuming that the content of centml_cred_file is always valid and that we don't have to check the content?
also, json.load can fail if the file it reads isn't json parsable, so maybe a try-except block is good?

Reply:
centml_cred_file is set by us. so normally it should always be valid and parsable. the only case is when the file is corrupt or the user intentionally modify it. but we can handle these cases later.

Reply:
makes sense :+1: